lib/rbac/filterer.rb

```diff
@@ -522,7 +522,8 @@ def scope_for_user_role_group(klass, scope, miq_group, user, managed_filters)
         if MiqUserRole != klass
           filtered_ids = pluck_ids(get_managed_filter_object_ids(scope, managed_filters))
           # Non admins can only see their own groups
-          scope = scope.with_groups(user.miq_group_ids) unless user_or_group.miq_user_role&.super_admin_user?
+          scope = scope.with_groups(user.miq_group_ids) unless user_or_group.miq_user_role&.super_admin_user? ||
+              user_or_group.miq_user_role&.tenant_admin_user?
```

Comment:
hmm, looking at this again, `unless` with a `||` always make me think more to grok it...

maybe this is more obvious:

```ruby
scope = scope.with_groups(user.miq_group_ids) if !user_or_group.miq_user_role&.super_admin_user? &&
!user_or_group.miq_user_role&.tenant_admin_user?
```

since the conditional is so long, maybe use a block:

```ruby
if !user_or_group.miq_user_role&.super_admin_user? && !user_or_group.miq_user_role&.tenant_admin_user?
  scope = scope.with_groups(user.miq_group_ids)
end
```

Reply:
maybe we can simplify this further by assigning role at the beginning of this else and use it in the 3 places that all do `user_or_group.miq_user_role`


```ruby

else
  role = user_or_group.miq_user_role
  ...

    if !role&.super_admin_user? && !role&.tenant_admin_user?
      scope = scope.with_groups(user.miq_group_ids)
    end
end